pkg/addons/seaweedfs/install.go

```diff
@@ -100,6 +100,21 @@ func createService(ctx context.Context, kcli client.Client, serviceCIDR string)
 
 	obj.ObjectMeta.Labels = ApplyLabels(obj.ObjectMeta.Labels, "s3")
 
+	var existingObj corev1.Service
+	if err := kcli.Get(ctx, client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}, &existingObj); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return errors.Wrap(err, "get s3 service")
+		}
+	} else {
+		if existingObj.Spec.ClusterIP != clusterIP {
+			existingObj.Spec.ClusterIP = clusterIP
+			if err := kcli.Update(ctx, &existingObj); err != nil {
+				return errors.Wrap(err, "update s3 service")
+			}
+		}
+		return nil
+	}
```

Comment:
I think you need to check if the current clusterIP matches the existing, and if it does not delete + recreate
IIRC changing the specified IP isn't a thing that can be done